sdk/resource/config.go

```diff
@@ -67,7 +55,24 @@ func (d detectAttributes) Detect(context.Context) (*Resource, error) {
 }
 
 // WithDetectors adds detectors to be evaluated for the configured resource.
+// Any use of WithDetectors disabled the default behavior, to reenable this
+// inlcude a WithDetectors(BuiltinDetectors...),
+// Examples:
+// `New(ctx)`: Use builtin `Detector`s.
+// `New(ctx, WithDetectors())`: Use no `Detector`s
+// `New(ctx, WithDetectors(NoOp{}))`: Use no `Detector`s
+// `New(ctx, WithDetectors(d1, d2))`: Use Detector `d1`, then overlay Detector `d2`
+// ```
+// New(ctx,
+//      WithDetectors(BuiltinDetectors...),
+//      WithDetectors(d1),
+// )
+// ```
+// Use The `BuiltinDetectors`, then overlay Detector `d1`
 func WithDetectors(detectors ...Detector) Option {
+	if len(detectors) == 0 {
+		return detectorsOption{detectors: []Detector{NoOp{}}}
```

Comment:
```suggestion
		detectors = []Detector{NoOp{}}
```

Unless there's a reason to not change `detectors` here I have a slight preference for a single return.